validation.go

```diff
@@ -149,3 +192,79 @@ func convertValidations(validations []Validation) []cpluginv1.ParameterValidatio
 	}
 	return out
 }
+
+// ApplyConfigValidations a utility function that applies all the validations for parameters, return an error that
+// consists of a combination of errors from the configurations.
+func applyConfigValidations(params map[string]Parameter, config map[string]string) error {
+	// cfg is the config map with default values assigned
+	cfg, multiErr := assignParamDefaults(params, config)
+
+	for pKey, param := range params {
+		if len(strings.TrimSpace(cfg[pKey])) != 0 {
+			multiErr = multierr.Append(multiErr, validateParamType(param, pKey, cfg[pKey]))
+		}
+		multiErr = multierr.Append(multiErr, validateParamValue(pKey, cfg[pKey], param))
```

Comment:
My OCD kicks in when I see two functions with the same arguments but in a different order 😬 🙈